File: Stacks/Patterns/Pattern4/Pattern4.py
stk=[]
lst=[1,5,2,11,6,4,5,11,10,100]
pse_lst=[0]*len(lst)

for i in range(0,len(lst)):
    while(len(stk)>0 and stk[-1]>=lst[i]):
        stk.pop()
    if len(stk)==0:
        pse_lst[i]=-1  
    else:
        pse_lst[i]=stk[-1]
    stk.append(lst[i])
print(pse_lst)

File: Stacks/Patterns/Pattern4/Pattern4.java


import java.util.*;
class Pattern3{
    public static void main(String[] args) {
        Stack<Integer> stk=new Stack<>();
        int[] num={5,2,7,3,6,3,2,10,66};
        int[] pse_num=new int[num.length];

        for (int i=0;i<num.length;i++){
            while(!stk.empty() && stk.peek()>=num[i]){
                stk.pop();
            }
            if(stk.empty()){
                pse_num[i]=-1;
            }
            else{
                pse_num[i]=stk.peek();
            }
            stk.push(num[i]);
        }
        System.out.println(Arrays.toString(pse_num));
    }
    
}

